sources/isal/igzip/adler32_base.h

```diff
@@ -0,0 +1,25 @@
+/*******************************************************************************
+ * Copyright (C) 2024 Intel Corporation
+ *
+ * SPDX-License-Identifier: MIT
+ ******************************************************************************/
+
+#ifndef QPL_SOURCES_ISAL_IGZIP_ADLER32_BASE_C
+#define QPL_SOURCES_ISAL_IGZIP_ADLER32_BASE_C
+
+// includes extracted from sources/isal/igzip/adler32_base.c
+#include <stddef.h>
+#include <stdint.h>
+#include "igzip_checksums.h"
+
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+    uint32_t qpl_adler32_base(uint32_t adler32, uint8_t *start, uint32_t length);
+#ifdef __cplusplus
+} // extern "C"
+#endif
```

Comment:
Unclear on this change goal. Is this header supposed to be included into `sources/isal/igzip/adler32_base.c` or to the `.c` or `.cpp` files where we're calling qpl_adler32_base (so that the declaration is readily available)?

For instance, this particular `qpl_adler32_base` function is used in a single `.asm` file that relies on extern, so probably not necessary to make its declaration available.
And it is also not required for `sources/isal/igzip/adler32_base.c` where it is being included currently.

Additionally, what is the reason for having these here -- 
...
#include <stddef.h>
#include <stdint.h>
#include "igzip_checksums.h"
... 
They are not required for using the `qpl_adler32_base` (this would probably require just including the header for uint32_t). And they are already present in `sources/isal/igzip/adler32_base.c`.

Reply:
+ similar comment for changes below.

In fact, these `foo_base` functions are an interesting case. They are supposed to be a reference implementation, and dispatching to either reference or optimized version is handled in some `.asm` as I mentioned earlier. So, really the `foo_base` would never be used directly, therefore no need for such declarations, the rest of the library would be using `foo` that is, in some sense, a wrapper that would dispatch to available optimization.

Reply:
The intent here (and similar below) is to be able to compile `adler32_base.c` with either a c or c++ compiler. If we use a c++ compiler, since there is no explicit declaration of language linkage, the symbol for `qpl_adler32_base` gets name-mangled, so we get a linker error since, as you pointed out, the callsite in the `asm` file sees it `extern` and expects the umangled symbol.

With this change included into the `.c` file that has the function definition, we're sure to always get C language linkage and an unmangled symbol.

You are right that, since the `asm` file never sees this declaration, we could fix the linkage inside the `.c` file instead of introducing this new header, if that is what you prefer.